pkg/testing/integration/program.go

```diff
@@ -1448,28 +1458,40 @@ func (pt *ProgramTester) copyTestToTemporaryDirectory() (string, string, error)
 	}
 	fprintf(pt.opts.Stdout, "pulumi: %v\n", bin)
 
-	// For most projects, we will copy to a temporary directory.  For Go projects, however, we must not perturb
-	// the source layout, due to GOPATH and vendoring.  So, skip it for Go.
+	stackName := string(pt.opts.GetStackName())
+
+	// For most projects, we will copy to a temporary directory.  For Go projects, however, we must create
+	// a folder structure that adheres to GOPATH requirements
 	var tmpdir, projdir string
 	if projinfo.Proj.Runtime.Name() == "go" {
-		projdir = projinfo.Root
+		// Ensure GOPATH is known.
+		gopath := os.Getenv("GOPATH")
+		if gopath == "" {
+			usr, userErr := user.Current()
+			if userErr != nil {
+				return "", "", errors.New("Couldn't get the current user")
+			}
+			gopath = filepath.Join(usr.HomeDir, "go")
+		}
+		folder := fmt.Sprintf("stackName-%d", time.Now().UnixNano())
+		tmpdir = filepath.Join(gopath, "src", folder)
+		err := os.Mkdir(tmpdir, 0700)
+		if err != nil {
```

Comment:
This looks like the same logic as added above - should we share this?

Reply:
Yeah I will refactor this to make it cleaner